### PR TITLE
Test lumos automatically on multiple versions of python

### PR DESCRIPTION
**Pull request by nicolasboisseau**
Initial version was tested on python 3.8
This change makes test run on 3.8,3.9,3.10 on linux and windows.

Each time a new pull request will be made, lumos setup and testing will be executed on the 3 python versions

### CODE REVIEW

**Comment by nicolasboisseau** [comment]
Hi @tlp19 ! I made a little summer change :) I try to put you in the loop to see lol !

**Comment by tlp19** [comment]
Hey @nicolasboisseau! Thanks for the update on the testing scripts.

All looks good to me, but I have a few questions about optimizing this workflow:
 - I believe you set the `max-parallel` parameter to 2 in [this commit](https://github.com/nicolasboisseau/lumos/commit/c25127c7a949e5e73617a001fb169201163bbc1e) on the `master` branch, but with the addition of testing on Python 3.10, could we not increase this to 3?
 - Also, I believe there might be a way to combine both the Windows and the Linux testing jobs into one single job, also using a `matrix` but with a second dimension (see [this guide](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-using-a-multi-dimension-matrix)  for reference). I don't mind trying this out myself, it would a nice refresher on Github Workflows for me, but you can also do it yourself directly if you prefer :)

In any case, all looks good to merge right now if you want, and I can change these things in a future PR.

**Review by tlp19 — APPROVED** [review]

**Comment by nicolasboisseau** [comment]
Hi! @tlp19 !!. Sure i played with the matrix and it was not so complex to setup. The main advantage is that is avoids code duplication. I even added macos testing to see and it works fine!. Concerning the max parallel argument, we could just remove it as github will automatically deal with parallel node counts automatically.

**Pull request merged**
